Shared/CMakeLists.txt

```diff
@@ -17,6 +13,10 @@ find_package(ZeroMQ REQUIRED)
 file(GLOB PROTO_SRC "Proto/ProtoCompiled/*.pb.cc")
 file(GLOB PROTO_HEADER "Proto/ProtoCompiled/*.pb.h")
 
+if(NOT MSVC)
+	set_source_files_properties(${PROTO_SRC} ${PROTO_HEADER} PROPERTIES COMPILE_FLAGS "-Wno-effc++ -Wno-unused-parameter")
```

Comment:
Seems to me like a wonderful way to avoid using ``pragma completely. Why not use that on all platforms for every external header?